src/qlever/commands/example_queries.py

```diff
@@ -212,89 +227,245 @@ def sparql_query_type(self, query: str) -> str:
         with open(queries_file, "r", encoding="utf-8") as q_file:
             try:
                 data = yaml.safe_load(q_file)  # Load YAML safely
             except yaml.YAMLError as exc:
                 log.error(f"Error parsing {queries_file} file: {exc}")
+                return []
 
-        error_msg = (
-            "Error: YAML file must contain a top-level 'queries' key."
-            "Error: 'queries' must be a list."
-            "Error: Each item in 'queries' must contain 'query' and 'sparql' keys."
-        )
         # Validate the structure
         if not isinstance(data, dict) or "queries" not in data:
-            log.error(error_msg)
-            return {}
+            log.error(
+                "Error: YAML file must contain a top-level 'queries' key"
+            )
+            return []
 
         if not isinstance(data["queries"], list):
-            log.error(error_msg)
-            return {}
+            log.error("Error: 'queries' key in YML file must hold a list.")
+            return []
 
         for item in data["queries"]:
             if (
                 not isinstance(item, dict)
                 or "query" not in item
                 or "sparql" not in item
             ):
-                log.error(error_msg)
-                return {}
+                log.error(
+                    "Error: Each item in 'queries' must contain "
+                    "'query' and 'sparql' keys."
+                )
+                return []
 
-        return data
+        # Get the list of query indices to keep
+        total_queries = len(data["queries"])
+        query_indices = []
+        for part in query_ids.split(","):
+            if "-" in part:
+                start, end = part.split("-")
+                if end == "$":
+                    end = total_queries
+                query_indices.extend(range(int(start) - 1, int(end)))
+            else:
+                idx = int(part) if part != "$" else total_queries
+                query_indices.append(idx - 1)
 
-    def get_example_queries(
+        try:
+            tsv_queries = []
+            for query_idx in query_indices:
+                if query_idx >= total_queries:
+                    log.error(
+                        "Make sure --query-ids don't exceed the total "
+                        "queries in the YML file"
+                    )
+                    return []
+                query = data["queries"][query_idx]
+
+                # Only include queries that match the query_regex if present
+                if query_regex:
+                    pattern = re.compile(query_regex, re.IGNORECASE)
+                    if not any(
+                        [
+                            pattern.search(query["query"]),
+                            pattern.search(query["sparql"]),
+                        ]
+                    ):
+                        continue
+
+                tsv_queries.append(f"{query['query']}\t{query['sparql']}")
```

Comment:
This code should come after the parsing, to avoid code duplication. The two parsers provide the same intermediate representation of the queries, and that one can then filter based on `query-ids` and `query-regex`.